source/MaterialXRender/Mesh.cpp

```diff
@@ -41,8 +42,17 @@ Mesh::Mesh(const string& name) :
 
 MeshStreamPtr Mesh::generateNormals(MeshStreamPtr positionStream)
 {
+    const string normalStreamName = "i_" + MeshStream::NORMAL_ATTRIBUTE;
+
+    // Return if already exists
+    MeshStreamPtr normalStream = getStream(normalStreamName, false);
+    if (normalStream)
+    {
+        return normalStream;
+    }
```

Comment:
If these methods were named `acquireNormals` and `acquireTangents` and used globally as vertex stream accessors, then I could see the value in adding these checks for existing data, but we currently use `generateNormals` and `generateTangents` with the goal of unconditionally generating new vertex streams.  Given that context, I'd recommend removing these checks from the core methods, and making sure that all of our callers have the same expectations from them.

Reply:
Actually all callers currently check for existence before calling these methods so this just places the logic inside the methods vs all callers having to add in the check.

Reply:
All true points, though I wouldn't recommend establishing a pattern where these checks occur *both* inside and outside these calls.

Since the names of these methods only suggest that data will be *generated*, and all clients currently check outside of the calls, perhaps the best approach is to omit the new inside checks?

Reply:
All true points, but I wouldn't recommend establishing a pattern where checks are performed *both* inside and outside these calls, as this reduces the clarity of where checks are truly needed.

Since the names of these methods only suggest that data will be *generated*, not cached and returned on future calls, and all clients currently perform checks for existing data, perhaps the best approach is to omit these new internal checks?